porcupine-core/src/Data/Representable.hs

```diff
@@ -6,8 +6,10 @@ import           Control.Applicative
 import           Data.Text
 
 
--- | A class for small objects that can be turned into a Text which will then be
--- stored in JSON String objects
+-- | A class for small objects that can be printed and read back from Text.
+--
+-- > fromTextRepr . toTextRepr == pure
+--
```

Comment:
Yes, String vs Text usage isn't extremely consistent in porcupine, for historical reasons. I wanted to clean all that, and possibly remove Representable in the meantime because it's mostly used to generate To/FromJSON instance.